### PR TITLE
CARDANO: dfns-1975: SDK integration Mesh + Blockfrost

### PR DESCRIPTION
**Pull request by Torakushi**
- [x] WAIT for PRs in DFNS monorepo to be merged

### CODE REVIEW

File: packages/lib-cardano-mesh/index.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  public get address(): string {
+    return this.metadata.address!
+
+  }
+
+  // Currently, we don't support signing arbitrary data
+  signData(address: string, payload: string) {
+    throw new Error('Method not implemented.')
```
Comment (djin-dfns) [inline packages/lib-cardano-mesh/index.ts]: let's throw `DfnsError` instead.

File: packages/lib-cardano-mesh/index.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import { DfnsApiClient, DfnsError } from '@dfns/sdk'
+import { GetWalletResponse, GenerateSignatureResponse } from '@dfns/sdk/types/wallets'
+
+import {ISigner} from '@meshsdk/core'
```
Comment (djin-dfns) [inline packages/lib-cardano-mesh/index.ts]: let's call our package `lib-meshsdk` to match with the package we are integrating. like how we did `taquito`.

File: examples/libs/cardano-mesh/basic-tx/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
+# Cardano basic transaction with Mesh and Blockfrost provider
+
+Demonstrates a very simple ADA transfer from a Dfns wallet to itself. Dfns wallet uses the well known mesh lib and blockfrost as a provider.
+
+See blockfros's [API Specification](https://blockfrost.dev/) and mesh doc (https://meshjs.dev/apis/transaction) for a complete guide.
```
Comment (djin-dfns) [inline examples/libs/cardano-mesh/basic-tx/README.md]: nit spelling error

```suggestion
See Blockfrost's [API Specification](https://blockfrost.dev/) and [Mesh documentation](https://meshjs.dev/apis/transaction) for a complete guide.
```

File: examples/libs/cardano-mesh/basic-tx/main.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    walletId: walletId,
+    dfnsClient,
+  })
+}
+
+const hexToBuffer = (hex: string): Buffer => {
+  return Buffer.from(hex.replace(/^0x/, ''), 'hex')
+}
+
+async function main() {
```
Comment (djin-dfns) [inline examples/libs/cardano-mesh/basic-tx/main.ts]: nit: for slightly better readability, maybe move the `main` function to below all helpers right above when it's called.

File: examples/libs/cardano-mesh/basic-tx/package.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "cardano-mesh-basic-tx",
+  "version": "1.0.0",
+  "main": "main.ts",
+  "dependencies": {
+    "@dfns/lib-cardano-mesh": "file:../../../packages/lib-cardano-mesh",
+    "@dfns/sdk": "file:../../../packages/sdk",
+    "@dfns/sdk-keysigner": "file:../../../packages/sdk-keysigner",
+    "@meshsdk/core": "1.5.10",
```
Comment (djin-dfns) [inline examples/libs/cardano-mesh/basic-tx/package.json]: add the @emurge dependency here.

File: examples/libs/cardano-mesh/basic-tx/package.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    "axios": "^1.6.7",
```
Comment (djin-dfns) [inline examples/libs/cardano-mesh/basic-tx/package.json]: don't think axios is needed.

File: package.json [hunk 7]
```diff
@@ -10,6 +10,7 @@
   },
   "dependencies": {
     "@aws-sdk/client-kms": "^3.485.0",
+    "@emurgo/cardano-serialization-lib-nodejs": "11.5.0",
```
Comment (djin-dfns) [inline package.json]: can remove. not a monorepo dependency, it's only directly referenced in the example

File: packages/lib-cardano-mesh/index.ts [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    })
+
+    assertSigned(res)
+    if (!res.signedData) {
+      throw new DfnsError(-1, 'signature missing', res)
+    }
+
+    return res.signedData
+  }
+}
```
Comment (djin-dfns) [inline packages/lib-cardano-mesh/index.ts]: nit: missing a end of file new line 😄

File: examples/libs/cardano-mesh/basic-tx/package.json (hunk 5 above)
Comment (Torakushi) [reply examples/libs/cardano-mesh/basic-tx/package.json]: It is already included in meshsdk:

```
"node_modules/@meshsdk/core": {
      "version": "1.5.10",
      "license": "Apache-2.0",
      "dependencies": {
        "@emurgo/cardano-message-signing-browser": "1.0.1",
        "@emurgo/cardano-message-signing-nodejs": "1.0.1",
        "@emurgo/cardano-serialization-lib-browser": "11.5.0",
        "@emurgo/cardano-serialization-lib-nodejs": "11.5.0",
        "@emurgo/cip14-js": "3.0.1",
        "axios": "0.27.2",
        "bip39": "3.0.4",
        "nanoid": "3.3.4",
        "zod": "3.19.1"
      },
      "engines": {
        "node": ">=16"
      }
    },
```

But I will add it so we can chose the right version

File: packages/lib-cardano-mesh/index.ts (hunk 8 above)
Comment (Torakushi) [reply packages/lib-cardano-mesh/index.ts]: added to my IDE setting :)

File: packages/lib-cardano-mesh/index.ts (hunk 1 above)
Comment (djin-dfns) [reply packages/lib-cardano-mesh/index.ts]: did you push up this change?

**Review by djin-dfns — APPROVED** [review]

**Pull request merged**